src/components/color-picker/color-picker-palette.tsx

```diff
@@ -0,0 +1,107 @@
+import { Component, h, Event, EventEmitter, Prop } from '@stencil/core';
+/**
+ * @private
+ */
+
+@Component({
+    tag: 'limel-color-picker-palette',
+    shadow: true,
+    styleUrl: 'color-picker-palette.scss',
+})
+export class Palette {
+    /**
+     * Color value that is manually typed by the user
+     */
+    @Prop()
+    public value: string;
+
+    /**
+     * Label of the input field.
+     */
+    @Prop()
+    public label: string;
+
+    /**
+     * elper text of the input field.
+     */
+    @Prop()
+    public helperText: string;
+
+    /**
+     * Emits chosen value to the parent component
+     */
+    @Event()
+    private change: EventEmitter<string>;
+
+    private colors = [
+        'red',
+        'pink',
+        'magenta',
+        'purple',
+        'violet',
+        'indigo',
+        'blue',
+        'sky',
+        'cyan',
+        'teal',
+        'green',
+        'lime',
+        'grass',
+        'yellow',
+        'amber',
+        'orange',
+        'coral',
+        'brown',
+        'gray',
+        'glaucous',
+    ];
+
+    private brightness = ['lighter', 'light', 'default', 'dark', 'darker'];
+
+    public render() {
+        const background = this.value ? { '--background': this.value } : {};
+
+        return [
+            <div class="color-picker-palette">{this.renderSwatches()}</div>,
+            <div class="chosen-color-name">
+                <limel-input-field
+                    label={this.label}
+                    helperText={this.helperText}
+                    value={this.value}
+                    onChange={this.handleChange}
+                />
+                <div class="chosen-color-preview" style={background} />
+            </div>,
+        ];
+    }
+
+    private renderSwatches = () => {
+        return this.colors.map((color) => {
+            return this.brightness.map(this.renderBrightness(color));
+        });
+    };
+    private renderBrightness = (color) => (brightnes) => {
```

Comment:
I was a bit confused when I helped you, but I think a better name for this could be `renderSwatch` since it is called from `renderSwatches` and renders a single swatch :slightly_smiling_face: